test/dlc_tests/sanity/test_dlc_labels.py

```diff
@@ -42,4 +44,71 @@ def test_dlc_major_version_label(image, region):
 
-    LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
+    test_utils.LOGGER.info(f"{image} has 'dlc_major_version' = {major_version}")
+
+
+@pytest.mark.integration("dlc_major_version_label")
+@pytest.mark.model("N/A")
+def test_dlc_version_dockerfiles(image):
+    """
+    Test to make sure semantic versioning scheme in Dockerfiles is correct
+
+    :param image: <str> ECR image URI
+    """
+    dlc_dir = os.getcwd().split(f"{os.sep}test{os.sep}")[0]
+    job_type = test_utils.get_job_type_from_image(image)
+    framework, fw_version = test_utils.get_framework_and_version_from_tag(image)
+    processor = test_utils.get_processor_from_image_uri(image)
+
+    root_dir = os.path.join(dlc_dir, framework, job_type, "docker")
+
+    # Skip older FW versions that did not use this versioning scheme
+    references = {"tensorflow2": "2.2.0", "tensorflow1": "1.16.0", "mxnet": "1.7.0", "pytorch": "1.5.0"}
+    if test_utils.is_tf1(image):
+        reference_fw = "tensorflow1"
+    elif test_utils.is_tf2(image):
+        reference_fw = "tensorflow2"
+    else:
+        reference_fw = framework
+    if processor != "eia" and fw_version < references[reference_fw]:
+        pytest.skip(
+            f"Not enforcing new versioning scheme on old image {image}. "
+            f"Started enforcing version scheme on the following: {references}"
+        )
+
+    dockerfiles = []
+    fw_version_major_minor = re.match(r"(\d+.\d+)", fw_version).group(1)
+    for root, dirnames, filenames in os.walk(root_dir):
+        for filename in filenames:
+            if filename == f"Dockerfile.{processor}":
+                dockerfile_path = os.path.join(root_dir, root, filename)
+                if "example" not in dockerfile_path and f"{os.sep}{fw_version_major_minor}" in dockerfile_path:
+                    dockerfiles.append(dockerfile_path)
+
+    versions = {}
+    dlc_label_regex = re.compile(r'LABEL dlc_major_version="(\d+)"')
+    for dockerfile in dockerfiles:
+        with open(dockerfile, "r") as df:
+            for line in df:
+                major_version_match = dlc_label_regex.match(line)
+                if major_version_match:
+                    versions[dockerfile] = int(major_version_match.group(1))
+
+    expected_versions = list(range(1, len(dockerfiles) + 1))
+    actual_versions = []
+
+    for _, version in versions.items():
+        actual_versions.append(version)
+
+    # Test case explicitly for TF2.3 gpu, since v1.0 is banned
+    if (framework, fw_version_major_minor, processor) == ("tensorflow", "2.3", "gpu"):
+        expected_versions = [v + 1 for v in expected_versions]
+        assert 1 not in actual_versions, (
+            f"DLC v1.0 is deprecated in TF2.3 gpu containers, but found major version 1 "
+            f"in one of the Dockerfiles. Please inspect {versions}"
+        )
+
+    assert sorted(actual_versions) == expected_versions, (
```

Comment:
From the logic of this test, we are expecting the dlc major versions of the Dockerfiles of 
1. a certain framework(e.g. tf) 
2. with a certain version x.y(e.g. 2.2) 
3. for a certain processor (e.g. gpu)

in a pattern of a consecutive set beginning with 1(e.g. {1, 2, 3}).

For example, there are 2 images for tf2.2 gpu. One is cu101 with major version 1, the other is cu102 with major version 2.

We are assuming every time we are going to increase the major version (for a certain framework version x.y and a certain processor, e.g. gpu), it introduces a new Dockerfile, and in the mean time, we are still retaining all the previous dockerfiles for it. However, there might be some other cases that break assumption.

According to our dlc version design, increment of major versions could also depend on “backward-compatibility”.
For example, we have cuda101 TF2.2 with major version 1 and cuda102 TF2.2 with major version 2 at the beginning. However, there is a breaking/backward-incompatible change of a dependency and we decide to upgrade the image for cuda102 TF2.2, then it should update it to major version 3 without introducing new Dockerfile. Now they would be in versions {1, 3} and breaks the assumed pattern.

Reply:
If we come across this scenario, though, shouldn't we still be displaying all of the major versions in Dockerfiles, so a user can build with a version that may have a breaking change?

Reply:
That is a good idea to display all existing versions information. However, the assertion would fail in this case but we still want to proceed with the given versions.
We can keep this test like this and allow the exception for it once there is such a scenario. A comment or an error prompt to remind release lead could be helpful.